File: 2943/2943.cpp
class Solution {
public:
    int maximizeAreaOfSquareHoleInGrid(TreeNode* root) {
        if(!root) return 0;
        return 1+max(maximizeAreaOfSquareHoleInGrid(root->left),maximizeAreaOfSquareHoleInGrid(root->right));
    }
};

File: 2943/2943.cs
public class Solution {
    public int MaximizeAreaOfSquareHoleInGrid(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximizeAreaOfSquareHoleInGrid(root.left), MaximizeAreaOfSquareHoleInGrid(root.right));
    }
}